src/core/reactor.cc

```diff
@@ -1921,6 +1925,191 @@ reactor::inotify_add_watch(int fd, std::string_view path, uint32_t flags) {
     });
 }
 
+future<std::tuple<pid_t, int, int, int>>
+reactor::spawn(std::string_view pathname,
+               std::vector<sstring> argv,
+               std::vector<sstring> env) {
+    return do_with(pid_t{},
+                   std::array<int, 2>{-1, -1},
+                   std::array<int, 2>{-1, -1},
+                   std::array<int, 2>{-1, -1},
+                   sstring(pathname),
+                   posix_spawn_file_actions_t{},
+                   posix_spawnattr_t{},
+                   std::move(argv),
+                   std::move(env),
+                   [this](auto& child_pid, auto& cin_pipe, auto& cout_pipe, auto& cerr_pipe, auto& pathname, auto& actions, auto& attr, auto& argv, auto& env) {
+        auto create_cin_pipe = _thread_pool->submit<syscall_result<int>>([&cin_pipe] {
```

Comment:
A suggestion (for next PR) -- it's worth having another API call (in namespace experimental). This place should become a one-line then.

Reply:
good point. will do so right after this change lands.

Reply:
#1300